src/control/server/drpc.go

```diff
@@ -45,6 +45,25 @@ func getDrpcClientConnection(sockDir string) *drpc.ClientConnection {
 	return drpc.NewClientConnection(clientSock)
 }
 
+func checkDrpcClientSocketPath(socketPath string) error {
+	if socketPath == "" {
+		return errors.New("Socket path empty")
+	}
+
+	f, err := os.Stat(socketPath)
+	if err != nil {
+		return errors.Errorf("Socket path '%s' could not be accessed: %s",
+			socketPath, err.Error())
+	}
+
+	if f.IsDir() {
```

Comment:
Probably better to check `if f.Mode()&os.ModeSocket == 0` here, right?

Reply:
Good catch, that's much better.

Reply:
Good catch, that's much better.